src/utils/hooks/useGetReviewInfo.tsx

```diff
@@ -0,0 +1,63 @@
+import { useEffect, useState } from 'react'
+import { AssignmentDetailsNEW } from '../types'
+import { Review, ReviewAssignment, ReviewStatus, useGetReviewInfoQuery } from '../generated/graphql'
+
+interface UseGetReviewInfoProps {
+  applicationId: number
+  userId: number
+}
+
+const useGetReviewInfo = ({ applicationId, userId }: UseGetReviewInfoProps) => {
+  const [assignments, setAssignments] = useState<AssignmentDetailsNEW[]>()
+
+  const { data, loading, error } = useGetReviewInfoQuery({
+    variables: {
+      reviewerId: userId,
+      applicationId,
+    },
+  })
+
+  useEffect(() => {
+    if (!data) return
+
+    const reviewAssigments = data.reviewAssignments?.nodes as ReviewAssignment[]
+    const assignments: AssignmentDetailsNEW[] = reviewAssigments.map((reviewAssignment) => {
+      // There will always just be one review assignment linked to a review.
+      const review = reviewAssignment.reviews.nodes[0] as Review
+      if (reviewAssignment.reviews.nodes.length > 1)
+        console.error(
+          'More then one review associated with reviewAssignment with id',
+          reviewAssignment.id
+        )
+
+      const stage = reviewAssignment.stage
+
+      // Extra field just to use in initial example - might conflict with future queries
+      // to get reviewQuestionAssignment
+      const totalAssignedQuestions = reviewAssignment.reviewQuestionAssignments.totalCount
+
+      const { id, status, timeCreated } = review
```

Comment:
it's quite possible that review will not exist

Reply:
Yes, very good point. I'll set it to `null` in case there is no review associated yet. I think that's how we have done in most places.

Reply:
Done: b68e7f2 & bbde101